lib/app/service/key_value_storage_service.dart

```diff
@@ -0,0 +1,702 @@
+      _storage.getDomain(domain: _seedsKey);
+
+  /// Add or rename public key
+  Future<void> addSeedOrRename({
+    required String masterKey,
+    required String name,
+  }) =>
+      _storage.set(masterKey, name, domain: _seedsKey);
+
+  /// Remove name of public key
+  Future<void> removeSeed(String masterKey) =>
+      _storage.delete(masterKey, domain: _seedsKey);
+
+  /// Clear information about all names of public keys
+  Future<void> clearSeeds() => _storage.clearDomain(_seedsKey);
+
+  /// Get current public key that user set before
+  Future<String?> get currentKey =>
+      _storage.get(_currentKey, domain: _preferencesKey);
+
+  /// Set current public key of user
+  Future<void> setCurrentKey(String publicKey) => _storage.set(
+        _currentKey,
+        publicKey,
+        domain: _preferencesKey,
+      );
+
+  /// Returns up to [maxLastSelectedSeeds] public keys of seeds that were used.
+  ///
+  /// After updating to application version with this list, it's filled with 4
+  /// (or less) random keys with [currentKey] at 1-st place.
+  Future<List<String>> lastViewedSeeds() async {
+    final seeds = await _storage.get(
+      _lastSelectedSeedsKey,
+      domain: _preferencesKey,
+    );
+    if (seeds == null) {
+      return [];
+    }
+    final seedsList = jsonDecode(seeds) as List<dynamic>;
+    return seedsList.cast<String>();
+  }
+
+  /// Update seeds that were used by user.
+  /// There must be only master keys, if key is sub, then put its master.
+  /// Count of seeds must be less or equals to [maxLastSelectedSeeds] and
+  /// cropped outside.
+  Future<void> updateLastViewedSeeds(List<String> seedsKeys) => _storage.set(
+        _lastSelectedSeedsKey,
+        jsonEncode(seedsKeys),
+        domain: _preferencesKey,
+      );
+
+  /// List of addresses of accounts that were hidden by user
+  Future<List<String>> get hiddenAccounts async {
+    final accounts = await _storage.get(
+      _hiddenAccountsKey,
+      domain: _preferencesKey,
+    );
+    if (accounts == null) {
+      return [];
+    }
+    final accountsList = jsonDecode(accounts) as List<dynamic>;
+    return accountsList.cast<String>();
+  }
+
+  /// Hide account address so it won't be displayed at accounts list
+  Future<void> hideAccount(String address) async {
+    final accounts = await hiddenAccounts;
+    accounts.add(address);
+    return _storage.set(
+      _hiddenAccountsKey,
+      jsonEncode(accounts.toSet().toList()),
+      domain: _preferencesKey,
+    );
+  }
+
+  /// Show account address so it will be displayed at accounts list
+  Future<void> showAccount(String address) async {
+    final accounts = await hiddenAccounts;
+    accounts.remove(address);
+    return _storage.set(
+      _hiddenAccountsKey,
+      jsonEncode(accounts),
+      domain: _preferencesKey,
+    );
+  }
+
+  /// Remove information about hidden account (make it visible)
+  Future<void> removeHiddenIfPossible(String address) async {
```

Comment:
What the difference between showAccount and removeHiddenIfPossible?

Reply:
-